tools/malwarejail/env/wscript.js

```diff
@@ -2201,3 +2183,62 @@ ga = function () {
 XPathResult = function () {
     return XPathResult;
 }
+
+// These constructors override the original class via aliasing.
+Blob = function() {
+    const _id = ++_object_id;
+    util_log("new Blob(" + _truncateOutput(Array.prototype.slice.call(arguments, 0).join(",") + ")"))
+    const blob_sources = arguments[0]
+    const blob_options = arguments[1]
+    var blob_filename = "Blob[" + _id + "]";
+    if ("filename" in blob_options) {
+        blob_filename = blob_options["filename"]
+        delete blob_options["filename"]
+    }
+    const blob = new BlobAlias(sources=blob_sources, options=blob_options)
+    blob.arrayBuffer().then(function(result) {
+        _wscript_saved_files[blob_filename] = Buffer.from(result);
+    })
+    return blob
+}
+
+URL = function() {
+    util_log("new URL(" + _truncateOutput(Array.prototype.slice.call(arguments, 0).join(",") + ")"))
+    return new URLAlias(arguments)
+}
+
+URL.createObjectURL = function () {
+    const _id = ++_object_id;
+    util_log("URL.createObjectURL(" + _truncateOutput(Array.prototype.slice.call(arguments, 0).join(",") + ")"))
+    const blob = arguments[0]
+    blob.arrayBuffer().then(function(result) {
```

Comment:
You just assume that it's always a blob? I believe in the original code, the "content" (arguments[0]) could be plain text, or something else?